src/components/ec/cuda/ec_cuda_executor_interruptible.c

```diff
@@ -48,8 +48,8 @@ ucc_cuda_executor_interruptible_task_post(ucc_ee_executor_t *executor,
                                          const ucc_ee_executor_task_args_t *task_args,
                                          ucc_ee_executor_task_t **task)
 {
+    cudaStream_t stream = NULL;
```

Comment:
That should fix build, but i don't think it's good solution in general. Event will be captured on a default stream creating false dependency. I think a better approach would be
* move ucc_cuda_executor_interruptible_get_stream to UCC_EE_EXECUTOR_TASK_TYPE_COPY section
* move ucc_ec_cuda_event_post after cudaMemcpyAsync in UCC_EE_EXECUTOR_TASK_TYPE_COPY
* change ucc_cuda_executor_interruptible_task_test accordingly i.e. do ucc_ec_cuda_event_test only if event was captured

Reply:
i don't get it. are you saying that after the call to "ucc_cuda_executor_interruptible_get_stream" the stream can still be unitialized? I thought that after line 55 the stream is always non-NULL - isn't this the case?

Reply:
Right, it's always initialized after get_stream, I just want to say maybe don't use event captured on this stream for reduce functions since they are running on a different streams actually.

Reply:
@Sergei-Lebedev  lets merge this as a build fix (since the issue currently blocks hpcx build) and plz prep a fix that you propose separately.